polaris/competition/_base.py

```diff
@@ -0,0 +1,229 @@
+from datetime import datetime
+import json
+from typing import Callable, TypeAlias, Union
+
+from pydantic import (
+    Field,
+    computed_field,
+    field_serializer,
+    field_validator,
+)
+
+from polaris.dataset._subset import Subset
+from polaris.evaluate import Metric
+from polaris.evaluate._predictions import CompetitionPredictions
+from polaris.experimental._dataset_v2 import DatasetV2
+from polaris.utils.dict2html import dict2html
+from polaris.utils.types import TargetType, TaskType, SplitType
+
+ColumnsType: TypeAlias = str | list[str]
+
+
+class CompetitionSpecification(DatasetV2):
+    """An instance of this class represents a Polaris competition. It defines fields and functionality
+    that in combination with the [`DatasetV2`][polaris.experimental._dataset_v2.DatasetV2] class, allow
+    users to participate in competitions hosted on Polaris Hub.
+
+    Examples:
+        Basic API usage:
+        ```python
+        import polaris as po
+
+        # Load the benchmark from the Hub
+        competition = po.load_competition("polaris/hello-world-competition")
+
+        # Get the train and test data-loaders
+        train, test = competition.get_train_test_split()
+
+        # Use the training data to train your model
+        # Get the input as an array with 'train.inputs' and 'train.targets'
+        # Or simply iterate over the train object.
+        for x, y in train:
+            ...
+
+        # Work your magic to accurately predict the test set
+        prediction_values = np.array([0.0 for x in test])
+        predictions = CompetitionPredictions(
+            name="first-prediction",
+            owner=HubOwner(slug="dummy-user"),
+            report_url="REPORT_URL",
+            target_labels=competition.target_cols,
+            test_set_labels=competition.test_set_labels,
+            test_set_sizes=competition.test_set_sizes,
+            predictions=prediction_values,
+        )
```

Comment:
I agree with you and that would be ideal, but what about all the other metadata which someone can apply to a prediction? E.g. `github_url`, `contributors`, `tags`, `user_attributes`, `description`. 

It could make the `submit_predictions` signature quite long if we added all of that. What do you think?

Reply:
Yeah, I agree, but I think it's the better option. 

We could also simplify the method signature with a single metadata object (rather than several parameters), but that just feels like an unnecessary level of indirection?

Reply:
Given that the primary way users will submit predictions is likely through that method, then I agree with having the longer signature. Our docs and tutorials will point to it too.